js/src/dropdown.js

```diff
@@ -241,16 +243,28 @@ const Dropdown = (() => {
       return placement
     }
 
+    _getOffset(data) {
+      const popperOffset = $.extend({}, data.offsets, this._config.offset(data.offsets) || {})
+      Object.keys(popperOffset).forEach((key) => {
+        data.offsets[key] = popperOffset[key]
+      })
+      return data
+    }
+
     _detectNavbar() {
       return $(this._element).closest('.navbar').length > 0
     }
 
     _getPopperConfig() {
+      let key = POPPER_OFFSET_KEY
+      if (typeof this._config.offset === 'function') {
+        key = POPPER_OFFSET_FN_KEY
+      }
```

Comment:
IMO something like that would be enough : 
```javascript
const offsetConf = {}
if (typeof this._config.offset === 'function') {
  offsetConf.offsetFn = (data) => this._getOffset(data)
} else {
  offsetConf.offset = this._config.offset
}

Reply:
I agree with your. It's better solution. Const is redundant
```javascript
_getPopperConfig() {
      const offsetConf = {}
      if (typeof this._config.offset === 'function') {
        offsetConf.fn = (data) => this._getOffset(data)
      } else {
        offsetConf.offset = this._config.offset
      }
      const popperConfig = {
        placement : this._getPlacement(),
        modifiers : {
          offset : offsetConf,
          flip : {
            enabled : this._config.flip
          }
        }
      }

      // Disable Popper.js for Dropdown in Navbar
      if (this._inNavbar) {
        popperConfig.modifiers.applyStyle = {
          enabled: !this._inNavbar
        }
      }
      return popperConfig
    }
```